### DISCUSSION TITLE
Clean Architecture wrong example

### DISCUSSION THREAD

**Discussion by Dotshift** [jasontaylordev/CleanArchitecture#795] [General]
I search alot an informations about clean architecture. But, It ends up tons of question. Cause, explanation with pratical use not sync. Even, the example template. 

if we look on [ApplicationDbContext file](https://github.com/jasontaylordev/CleanArchitecture/blob/main/src/Infrastructure/Persistence/ApplicationDbContext.cs). At line 3, this file dependent on domain assembly. It same with [Entity Config file](https://github.com/jasontaylordev/CleanArchitecture/blob/main/src/Infrastructure/Persistence/Configurations/TodoListConfiguration.cs). It totally broke the concept Clean Architecture that i understand. 

Clean Architecture have 4 layers, Domain, Application, Infrastructure, UI.  Domain don't have depedency, Application dependent only Domain. Infrastructure dependent only on Application. So in this example solution, Infrastructure should dependent on medium class in Application assembly not Domain assembly.  Other article that discuss about this [Architecture](https://medium.com/dotnet-hub/clean-architecture-with-dotnet-and-dotnet-core-aspnetcore-overview-introduction-getting-started-ec922e53bb97) with picture. 
![1_Vk7quy-rCWYom9kJ00V4sw](https://user-images.githubusercontent.com/81574901/218147525-9f578423-bf6d-4015-8cbc-34fe65090f1f.jpg)

is there something that i missing here?

**Comment by akshay-zz** [comment]
I don't see things as this is the hard rule to create software. Software development always going to be opinionated and different for different people.  You could have pointed out also that our Domain is dependent on `MediatR` . This also violates the rule. It's all about does it solving your problem or not and whether the pros are above the cons or not. 
We can take one more example there's the type that is specific to a database. An example could be the `Geometry` type in Postgres. But to use it your domain needs to be dependent on database specific type that will violet clean code again. So for me, the theory is there to guide you but in the practical world, the pendulum will always going to shift. One architecture or structure can't solve all the problems. Hopefully, this makes sense.

**Reply by prince272** [reply]
Exactly. I had that same problem when adding identity to the clean architecture. When setting up identity, you need customize identity classes (IdentityUser, IdentityRole) in the Domain project but the identity classes comes from an external package.
Well, you might be tempted to add the classes inside the infrastructure project, but what if some domain entities depends on that class (User?).
